index.js

```diff
@@ -162,8 +162,14 @@ async function findPlugins (dir, options, hookedAccumulator = {}, prefix, depth
   if (indexDirent) {
     const file = path.join(dir, indexDirent.name)
     const type = getScriptType(file, options.packageType)
-    if (type === 'typescript' && !typescriptSupport) {
-      throw new Error(`@fastify/autoload cannot import hooks plugin at '${file}'. To fix this error compile TypeScript to JavaScript or use 'ts-node' to run your app.`)
+    if (type === 'typescript') {
+      if (!typescriptSupport) {
```

Comment:
nesting two ifs this way is the same as what it was before, why not just keep it as it was?